maven-plugin/src/main/java/org/jvnet/localizer/Generator.java

```diff
@@ -65,45 +73,85 @@ protected void generateImpl(String className, Properties props) throws Assertion
         try {
             JDefinedClass c = cm._class(className);
             c.annotate(SuppressWarnings.class).paramArray("value").param("").param("PMD").param("all");
+            c.javadoc().add("Generated localization support class.");
 
             // [RESULT]
             // private static final ResourceBundleHolder holder = BundleHolder.get(Messages.class);
 
-            JVar holder = c.field(JMod.PRIVATE | JMod.STATIC | JMod.FINAL, ResourceBundleHolder.class, "holder",
+            JFieldVar holder = c.field(JMod.PRIVATE | JMod.STATIC | JMod.FINAL, ResourceBundleHolder.class, "holder",
                     cm.ref(ResourceBundleHolder.class).staticInvoke("get").arg(c.dotclass()) );
+            holder.javadoc().add("The resource bundle reference");
 
 
             for (Entry<Object,Object> e : props.entrySet()) {
                 String key = e.getKey().toString();
                 String value = e.getValue().toString();
 
-                int n = countArgs(value);
+                final Format[] formats = new MessageFormat(value).getFormatsByArgumentIndex();
+                Map<String,String> params = new HashMap<String, String>();
```

Comment:
Does the order of the `params` matter? Kinda hard to tell from when viewing the diff in github, but it looks like it might coz it looks like `params` is used to add the javadoc for the parameter list.